src/change_stream/mod.rs

```diff
@@ -0,0 +1,172 @@
+//! Contains the functionality for ChangeStreams.
+pub mod document;
+pub mod option;
+
+use std::{marker::PhantomData, pin::Pin, task::Poll};
+
+use bson::Document;
+use futures::{stream::Stream, task::Context};
+use serde::{de::DeserializeOwned, Deserialize};
+
+use crate::{
+    change_stream::{
+        document::{ChangeStreamEventDocument, ResumeToken},
+        option::ChangeStreamOptions,
+    },
+    error::Result,
+    options::AggregateOptions,
+    selection_criteria::ReadPreference,
+    Client,
+    Collection,
+    Cursor,
+    Database,
+};
+
+/// A `ChangeStream` streams the ongoing changes of its associated collection, database or
+/// deployment. `ChangeStream` instances should be created with method `watch` against the relevant
+/// target.
+/// `ChangeStream`'s are "resumable", meaning that they can be restarted at a certain place in the
+/// stream of events. This is done automatically when the ChangeStream encounters certain
+/// "resumable" errors, such as transient network failures. It can also be done manually by passing
+/// a ResumeToken retrieved from a past event into either the resume_after or start_after (4.2+)
+/// options used to create the ChangeStream. Issuing a raw change stream aggregation is discouraged
+/// unless users wish to explicitly opt out of resumability.
+///
+/// A `ChangeStream` can be iterated to return batches of instances of any type that implements the
+/// Deserialize trait. By default, it returns batches of `ChangeStreamEventDocument`. These
+/// documents correspond to changes in the associated collection, database or deployment.
+///
+/// A change stream can be iterated like any other `Iterator`:
+///
+/// ```rust
+/// #[cfg(not(feature = "sync"))]
+/// # use futures::stream::StreamExt;
+/// # use mongodb::{Client, error::Result, bson::doc,
+/// # change_stream::document::ChangeStreamEventDocument};
+/// # #[cfg(feature = "async-std-runtime")]
+/// # use async_std::task;
+/// # #[cfg(feature = "tokio-runtime")]
+/// # use tokio::task;
+/// #
+/// # #[cfg(not(feature = "sync"))]
+/// # async fn func() -> Result<()> {
+/// # let client = Client::with_uri_str("mongodb://example.com").await?;
+/// # let coll = client.database("foo").collection("bar");
+/// let mut change_stream = coll.watch(None, None).await?;
+/// let result = coll.insert_one(doc! { "x": 1 }, None).await?;
+/// while let Some(change) = change_stream.next().await {
+///     let document: ChangeStreamEventDocument = change?;
+///     let full_document =
+///     println!("operation performed: {:?}, document: {:?}", document.operation_type, document.full_document);
+///     // operation performed: Insert, document: { "x": 1 }
```

Comment:
the print would actually look like `operation performed: Insert, document: Some(Document({"x": Int32(1)}))`

Reply:
fixed